toolchain/check/import_cpp.cpp

```diff
@@ -339,6 +513,52 @@ static auto MapType(Context& context, clang::QualType type) -> TypeExpr {
           .type_id = SemIR::ErrorInst::TypeId};
 }
 
+// Maps a C++ record type to a Carbon type.
+// TODO: Support more record types.
+static auto MapRecordType(Context& context, SemIR::LocId loc_id,
+                          const clang::RecordType& type) -> TypeExpr {
+  auto* record_decl = clang::dyn_cast<clang::CXXRecordDecl>(type.getDecl());
+  if (record_decl && record_decl->isStruct()) {
+    auto& clang_decls = context.sem_ir().clang_decls();
+    auto struct_clang_decl_id = clang_decls.Lookup(
+        {.decl = record_decl, .inst_id = SemIR::InstId::None});
+    if (!struct_clang_decl_id.has_value()) {
+      auto parent_inst_id =
+          MapDeclContext(context, record_decl->getDeclContext());
+      auto parent_name_scope_id =
+          context.insts().GetAs<SemIR::Namespace>(parent_inst_id).name_scope_id;
+      SemIR::InstId struct_inst_id = ImportCXXRecordDecl(
```

Comment:
Is there any risk of this call and the one in `ImportNameDecl` duplicating work, and importing the same class twice?

Reply:
Nice catch!

If `ImportNameDecl()` imports a struct, it will add it `clang_decls` so it won't be imported by `MapRecordType()`.
If `MapRecordType()` imports a struct, it indeed doesn't add its name, so we still import it when lookup by name afterwards.

To fix this:
1. I've removed the unnecessary attempt to add to `clang_decls` after calling `ImportCXXRecordDecl()`, which did nothing.
2. I've added tests.
3. I've added the name to the scope in this case.